drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -4038,6 +4038,23 @@ void RasterizerSceneGLES3::decals_set_filter(RS::DecalFilter p_filter) {
 void RasterizerSceneGLES3::light_projectors_set_filter(RS::LightProjectorFilter p_filter) {
 }
 
+bool RasterizerSceneGLES3::get_material_has_transparency(RID p_material_rid) const {
+	GLES3::SceneMaterialData *material = dynamic_cast<GLES3::SceneMaterialData *>(GLES3::MaterialStorage::get_singleton()->material_get_data(p_material_rid, RenderingServer::SHADER_SPATIAL));
+	// Can't found the SceneMaterialData corresponding to p_material_rid
+	if (!material)
+		return false;
+
+	bool has_read_screen_alpha = material->shader_data->uses_screen_texture || material->shader_data->uses_depth_texture || material->shader_data->uses_normal_texture;
+	bool has_base_alpha = ((material->shader_data->uses_alpha && !material->shader_data->uses_alpha_clip) || has_read_screen_alpha);
+	bool has_blend_alpha = material->shader_data->uses_blend_alpha;
+	bool has_alpha = has_base_alpha || has_blend_alpha;
+
+	if (has_alpha || has_read_screen_alpha || material->shader_data->depth_draw == GLES3::SceneShaderData::DEPTH_DRAW_DISABLED || material->shader_data->depth_test == GLES3::SceneShaderData::DEPTH_TEST_DISABLED) {
+		return true;
+	}
+	return false;
+}
+
```

Comment:
Based on : https://github.com/godotengine/godot/blob/master/drivers/gles3/rasterizer_scene_gles3.cpp#L200

I could have used my new method in place of the original code, but I didn't want to add regressing in a bugfix.